brave/brave-core
# Fixes balance drop on android

Resolves https://github.com/brave/brave-browser/issues/9992
Resolves https://github.com/brave/brave-browser/issues/10471

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
Plan 1:
1) start browser on master
2) enable rewards
3) claim grant
4) close browser
5) open DB and change:
   - `type` in promotion table from 0 to 1
   - empty unblinded_tokens table
6) start browser and make sure that balance is 0
7) close browser
8) switch to this PR
9) start browser
10) make sure that balance is showing correct again

Plan 2:
1) start browser on master
2) enable rewards
3) claim grant
4) make a tip
4) close browser
5) open DB and change:
   - `type` in promotion table from 0 to 1
   - empty unblinded_tokens table
6) start browser and make sure that balance is 0
7) close browser
8) switch to this PR
9) start browser
10) make sure that balance is still showing 0
11) open logs and make sure that you see this request and success response for it
```
[ REQUEST ]
> URL: https://grant.rewards.brave.software/v1/wallets/{paymentId}/events/batloss/1
> Method: UrlMethod::PUT
> Content: {"amount": 19.000000}
> Content Type: application/json; charset=utf-8
```

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
